File: Python/kids_with_candies.py
# Added
from typing import List

# Solution
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        # Max value in candies
        greatest = max(candies)
        # The following is a 'generator expression'
        # Note the difference with list comprehensions: parenthesis instead of square brackets
        # This creates a generator without the yield keyword
        # In a real-world scenario, it is recommended to convert the generator object into a list
        # The above will ensure that the return type will match to what's actually returned
        # Either use "[]" or list((candy + extraCandies >= greatest for candy in candies))
        return (candy + extraCandies >= greatest for candy in candies)

File: C++/kids_with_candies.cpp
// Added
// #include <string>   // No strings as far as the eye can C
// #include <iostream> // If you want to test it
#include <vector>
#include <algorithm> // max_element()
using namespace std;

// Solution
class Solution
{
public:
    vector<bool> kidsWithCandies(vector<int> &candies, int &extraCandies)
    {
        // Get the max element of a vector
        int greatest = *max_element(begin(candies), end(candies));
        vector<bool> res;
        // Iter once and push true|false accordingly
        for (int &c : candies)
            res.push_back((c + extraCandies) >= greatest);
        return res;
    }
};
